### PR TITLE
Added analytics hook support

### PR DESCRIPTION
**Pull request by dbotha**

### CODE REVIEW

File: KitePrintSDK/src/main/java/ly/kite/analytics/Analytics.java [hunk 1]
```diff
@@ -356,6 +358,36 @@ private Analytics( Context context )
...
+      // Load registered analytics event callback (if any) via reflection
+      String callbackClassName = mContext.getString(R.string.analytics_event_callback_class_name);
+      try
+        {
+        Class<?> clazz = Class.forName( callbackClassName );
+        Constructor<?> ctor = clazz.getConstructor( Context.class );
+        mCachedEventCallback = (IAnalyticsEventCallback) ctor.newInstance( new Object[] { mContext } );
+      } catch ( Exception ex )
+        {
+        mCachedEventCallback = null;
```
Comment (ghost) [inline KitePrintSDK/src/main/java/ly/kite/analytics/Analytics.java]: Wouldn't it be better to set the callback to the null callback? That way you don't have to check for null in every method further down.

File: KitePrintSDK/src/main/java/ly/kite/analytics/IAnalyticsEventCallback.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/*****************************************************
+ *
+ * TemplateClass.java
```
Comment (ghost) [inline KitePrintSDK/src/main/java/ly/kite/analytics/IAnalyticsEventCallback.java]: I always forget to change this too!

File: KitePrintSDK/src/main/java/ly/kite/analytics/Analytics.java (hunk 1 above)
Comment (dbotha) [inline KitePrintSDK/src/main/java/ly/kite/analytics/Analytics.java]: Good point, will do that now. I added the Null callback as a last minute thing just to check it all works as expected and entirely forgot I did intend to remove those null checks.

**Comment by dbotha** [comment]
@JLKite  Issue raised in comments are now addressed

**Pull request merged**